xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1254,157 +1325,67 @@ HDR_STATUS CWIN32Util::ToggleWindowsHDR(DXGI_MODE_DESC& modeDesc)
     }
   }
 #else
-  uint32_t pathCount = 0;
-  uint32_t modeCount = 0;
+  std::wstring gdiDeviceName{GetCurrentDisplayName()};
+  if (gdiDeviceName.empty())
+    return HDR_STATUS::HDR_TOGGLE_FAILED;
 
-  MONITORINFOEXW mi = {};
-  mi.cbSize = sizeof(mi);
-  GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi);
-  const std::wstring deviceNameW = mi.szDevice;
+  const auto identifier{GetDisplayTargetId(gdiDeviceName)};
+  if (!identifier)
+    return HDR_STATUS::HDR_TOGGLE_FAILED;
 
-  if (ERROR_SUCCESS != GetDisplayConfigBufferSizes(QDC_ONLY_ACTIVE_PATHS, &pathCount, &modeCount))
-    return status;
+  status = GetDisplayHDRStatus(identifier.value());
+  if (status == HDR_STATUS::HDR_UNSUPPORTED)
+    return HDR_STATUS::HDR_TOGGLE_FAILED;
 
-  std::vector<DISPLAYCONFIG_PATH_INFO> paths(pathCount);
-  std::vector<DISPLAYCONFIG_MODE_INFO> modes(modeCount);
-
-  if (ERROR_SUCCESS != QueryDisplayConfig(QDC_ONLY_ACTIVE_PATHS, &pathCount, paths.data(),
-                                          &modeCount, modes.data(), nullptr))
-    return status;
+  status = SetDisplayHDRStatus(identifier.value(), status == HDR_STATUS::HDR_OFF ? true : false,
+                               gdiDeviceName, modeDesc);
+#endif
+  return status;
+}
 
-  DISPLAYCONFIG_SOURCE_DEVICE_NAME getSourceName = {};
-  getSourceName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
-  getSourceName.header.size = sizeof(getSourceName);
+#ifndef TARGET_WINDOWS_STORE
+HDR_STATUS CWIN32Util::SetDisplayHDRStatus(CWIN32Util::DisplayConfigId identifier,
+                                           bool enable,
+                                           std::wstring gdiDeviceName,
+                                           DXGI_MODE_DESC& modeDesc)
```

Comment:
Parameters gdiDeviceName and modeDesc are not necessary  if you keep code the bloc

`// Restores previous graphics mode before toggle HDR`

in the ToggleWindowsHDR method

Reply:
Technically yes because Toggle is the public method, but logically it's related to setting the hdr status on / off.
I'll make the change but that part bothers me anyway, For encapsulation Set should be able to retrieve the current resolution by itself and restore it at the end (even if it's a little less efficient).